tests/functional/test_inference.py

```diff
@@ -84,51 +83,62 @@ def test_smc():
     smc = elfi.SMC(m['d'], batch_size=20000)
     res = smc.sample(N, thresholds=thresholds)
 
-    check_inference_with_informative_data(res, N, true_params)
+    check_inference_with_informative_data(res.samples, N, true_params)
 
     # We should be able to carry out the inference in less than six batches
     assert res.populations[-1].n_batches < 6
 
 
 @slow
 @pytest.mark.usefixtures('with_all_clients')
-def test_bayesian_optimization():
+def test_BOLFI():
```

Comment:
I think we should have a separate test for bayesian optimization. Now the method is untested. Maybe `test_bayesian_optimization_and_bolfi`. I was thinking that first run the BO test as it was but then just give the GP model to BOLFI and do the sampling with BOLFI. Then a separate quick test for running everything with BOLFI.